worlds/mlss/Client.py

```diff
@@ -0,0 +1,266 @@
+        except UnicodeDecodeError:
+            return False
+        except bizhawk.RequestFailedError:
+            return False  # Should verify on the next pass
+
+        ctx.game = self.game
+        ctx.items_handling = 0b101
+        ctx.want_slot_data = True
+        ctx.watcher_timeout = 0.125
+        self.rom_slot_name = rom_name
+        name_bytes = ((await bizhawk.read(ctx.bizhawk_ctx, [(0xB0, 16, "ROM")]))[0])
+        player = ((await bizhawk.read(ctx.bizhawk_ctx, [(0xAF, 1, "ROM")]))[0][0])
+        self.player = player + 1
+        name = bytes([byte for byte in name_bytes if byte != 0]).decode("UTF-8")
+        self.player_name = name
+        logger.info(name)
+
+        for i in range(59):
+            self.checked_flags[i] = [9, 10]
+
+        return True
+
+    async def set_auth(self, ctx: BizHawkClientContext) -> None:
+        ctx.auth = self.player_name
+
+    async def game_watcher(self, ctx: BizHawkClientContext) -> None:
+        from CommonClient import logger
+        try:
+            read_state = await bizhawk.read(ctx.bizhawk_ctx, [(0x4564, 59, "EWRAM"),
+                                                              (0x2330, 2, "IWRAM"), (0x3FE0, 1, "IWRAM"), (0x304A, 1, "EWRAM"), (0x304B, 1, "EWRAM"), (0x304C, 4, "EWRAM"), (0x4800, 6, "EWRAM"), (0x4808, 2, "EWRAM"), (0x4407, 1, "EWRAM"), (0x2339, 1, "IWRAM")])
+            flags = read_state[0]
+            room = (read_state[1][1] << 8) + read_state[1][0]
+            shop_init = read_state[2][0]
+            shop_scroll = read_state[3][0] & 0x1F
+            is_buy = (read_state[4][0] != 0)
+            shop_address = (struct.unpack('<I', read_state[5])[0]) & 0xFFFFFF
+            logo = bytes([byte for byte in read_state[6] if byte < 0x70]).decode("UTF-8")
+            received_index = (read_state[7][0] << 8) + read_state[7][1]
+            cackletta = (read_state[8][0] & 0x40)
+            shopping = (read_state[9][0] & 0xF)
+
+            if logo != "MLSSAP":
+                return
+
+            locs_to_send = set()
+            location = 0
+
+            # Checking shop purchases
+            if is_buy:
+                await bizhawk.write(ctx.bizhawk_ctx, [(0x304A, [0x0, 0x0], "EWRAM")])
+                if shop_address != 0x3c0618 and shop_address != 0x3c0684:
+                    location = shop[shop_address][shop_scroll]
+                else:
+                    if shop_init & 0x1 != 0:
+                        location = badge[shop_address][shop_scroll]
+                    else:
+                        location = pants[shop_address][shop_scroll]
+                if location in ctx.server_locations:
+                    locs_to_send.add(location)
+
+            for i in range(len(ctx.items_received) - received_index):
+                item_data = items_by_id[ctx.items_received[received_index + i].item]
+                b = await bizhawk.read(ctx.bizhawk_ctx, [(0x3057, 1, "EWRAM")])
+                await asyncio.sleep(.05)
```

Comment:
Would be nice to have a comment explaining this loop. Took me a minute to figure out what the goal was and why you're sleeping here.